src/ansys/dpf/post/selection.py

```diff
@@ -0,0 +1,557 @@
+            self._selection.set_input_name(_WfNames.streams, forward_sc.inputs.any)
+            for pin, ns in enumerate(named_selection):
+                mesh_scoping_op = operators.scoping.on_named_selection(
+                    requested_location=location,
+                    named_selection_name=ns,
+                    server=self._server,
+                    # data_sources=forward_ds.outputs.any,
+                    # streams_container=forward_sc.outputs.any,
+                )
+                mesh_scoping_op.connect(3, forward_sc.outputs.any)
+                mesh_scoping_op.connect(4, forward_ds.outputs.any)
+                op.connect(pin, mesh_scoping_op.outputs.mesh_scoping)
+            self._selection.set_output_name(_WfNames.scoping, op.outputs.merged_scoping)
+
+        self._selection.add_operator(op)
+
+    def select_with_scoping(self, scoping: Scoping):
+        """Directly sets the scoping as the spatial selection.
+
+        Parameters
+        ----------
+        scoping:
+            Scoping to use for spatial selection.
+        """
+        if not isinstance(scoping, Scoping):
+            raise TypeError(
+                f"The input scoping is an instance of {str(type(scoping))} "
+                f"instead of an expected {str(Scoping)}."
+            )
+
+        op = operators.utility.forward(scoping, server=self._server)
+        self._selection.add_operator(op)
+        self._selection.set_output_name(_WfNames.scoping, op.outputs.any)
+
+    def select_nodes(self, nodes: Union[List[int], Scoping]) -> None:
+        """Select nodes using their IDs or a nodal mesh scoping.
+
+        Parameters
+        ----------
+        nodes :
+            node IDs or nodal mesh scoping.
+        """
+        if isinstance(nodes, Scoping):
+            scoping = nodes
+        else:
+            scoping = Scoping(location=locations.nodal, ids=nodes, server=self._server)
+        self.select_with_scoping(scoping)
+
+    def select_nodes_of_elements(
+        self,
+        elements: Union[List[int], Scoping],
+        mesh: Mesh,
+    ) -> None:
+        """Select all nodes of elements using the elements' IDs or an elemental mesh scoping.
+
+        Parameters
+        ----------
+        elements:
+            element IDs or elemental mesh scoping.
+        mesh:
+            Mesh containing the necessary connectivity.
+        """
+        if isinstance(elements, Scoping):
+            scoping = elements
+        else:
+            scoping = Scoping(
+                location=locations.elemental, ids=elements, server=self._server
+            )
+
+        op = operators.scoping.transpose(
+            mesh_scoping=scoping, meshed_region=mesh._meshed_region, inclusive=0
+        )
+        self._selection.add_operator(op)
+        self._selection.set_output_name(
+            _WfNames.scoping, op.outputs.mesh_scoping_as_scoping
+        )
```

Comment:
If we are aiming to select the nodes of some elements.... shouldn't you at some point to switch to ``location.nodal``??

Reply:
That is what the transpose operator does.